go-controller/pkg/ovn/gateway_cleanup.go

```diff
@@ -128,15 +128,20 @@ func delPbrAndNatRules(nodeName string) {
 			"stdout: %s, stderr: %s, error: %v", matches, stderr, err)
 		matches = ""
 	}
-	nodeSubnetMatchSubStr := fmt.Sprintf("rtos-%s", nodeName)
+	// matches inport for a policy, must use ending quote to substring match to avoid matching other nodes accidentally
+	// i.e. rtos-ovn-worker would match rtos-ovn-worker2
+	nodeSubnetMatchSubStr := fmt.Sprintf("rtos-%s\"", nodeName)
+	// for inter node, match the comment in the policy, and include extra space to avoid accidental submatch
+	interNodeMatchSubStr := fmt.Sprintf("inter-%s ", nodeName)
```

Comment:
likewise include the space before.

except also if someone has nodes "foo" and "inter-foo" then the `interNodeMatchSubStr` for node "foo" would match the management port policy rule for node "inter-foo". (Though I guess since the priority would be wrong we wouldn't actually successfully delete it, we'd just log a confusing error? So probably OK if this code is eventually going away.)

Reply:
yeah someone would have to name their node inter-<worker name> and have a name foo. I think its OK to ignore this case.